### PR TITLE
(PC-24594) feat(EmailResend): add timer in resend email modal

### PR DESCRIPTION
**Pull request by marineshaw**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-24594

## Checklist

I have:

- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [x] Written **unit tests** native (and web when implementation is different) for my feature.
- [x] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]

## Screenshots

**delete** _if no UI change_

| Platform         | Mockup | After |
| :--------------- | :-----------: | :---: |
| iOS              | image |  image |
| Android          |               |       |
| Phone - Chrome   |               |image|
| Desktop - Chrome |               |image|

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/libs/hooks/useTimer.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from 'react'
+
+export const useTimer = (initialValue: number) => {
```
Comment (aliraiki) [inline src/libs/hooks/useTimer.ts:3]:
```suggestion
export const useTimer = (initialValueInSeconds: number) => {
```

File: src/libs/hooks/useTimer.native.test.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { useTimer } from 'libs/hooks/useTimer'
+import { renderHook, act } from 'tests/utils'
+
+jest.useFakeTimers({ legacyFakeTimers: true })
+
+describe('useTimer', () => {
+  it('should descrease every second', async () => {
```
Comment (aliraiki) [inline src/libs/hooks/useTimer.native.test.ts:7]:
```suggestion
  it('should decrease every second', async () => {
```

File: src/libs/hooks/useTimer.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  const [timeLeft, setTimeLeft] = useState(initialValue)
+
+  useEffect(() => {
+    let timerId: NodeJS.Timeout
```
Comment (aliraiki) [inline src/libs/hooks/useTimer.ts:7]: Je ne sais pas si c'est mieux de faire comme ça ou d'utiliser `useRef` 🤔

**Review by aliraiki — APPROVED** [review]

File: src/libs/hooks/useTimer.ts (hunk 3 above)
Comment (marineshaw) [reply src/libs/hooks/useTimer.ts:7]: Tu l'imagines comment avec useRef ?

Comment (aliraiki) [reply src/libs/hooks/useTimer.ts:7]: Tu peux regarder dans `useWhiteStatusBarWithoutReactNavigation` par exemple

**Review by aliraiki — APPROVED** [review]

**Review by Lucasbeneston — APPROVED** [review]

**Pull request merged**